### ISSUE TITLE
OutOfMemoryError during the AppIntro

### ISSUE THREAD

**Issue by jcgruenhage** [jcgruenhage/PlayMusicExporter#6]
There have been multiple reports of `java.lang.OutOfMemoryError` errors, that cause the application to crash.

One of the stack traces:

```
---------- Information -----------
PackageName: re.jcg.playmusicexporter
Crashed activity: activities.MusicContainerListActivity
----------- Exception ------------
Failed to allocate a 45813444 byte allocation with 718456 free bytes and 701KB until OOM
java.lang.OutOfMemoryError at 
dalvik.system.VMRuntime.newNonMovableArray(Native Method)
android.graphics.BitmapFactory.nativeDecodeAsset(Native Method)
android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:620)
android.graphics.BitmapFactory.decodeResourceStream(BitmapFactory.java:455)
android.graphics.drawable.Drawable.createFromResourceStream(Drawable.java:1155)
android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:720)
android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:571)
android.content.res.Resources.getDrawable(Resources.java:771)
android.content.Context.getDrawable(Context.java:525)
android.support.v4.content.ContextCompatApi21.getDrawable(Unknown Source)
android.support.v4.content.ContextCompat.getDrawable(Unknown Source)
android.support.v7.widget.AppCompatDrawableManager.getDrawable(Unknown Source)
android.support.v7.widget.AppCompatDrawableManager.getDrawable(Unknown Source)
android.support.v7.content.res.AppCompatResources.getDrawable(Unknown Source)
android.support.v7.widget.AppCompatImageHelper.setImageResource(Unknown Source)
android.support.v7.widget.AppCompatImageView.setImageResource(Unknown Source)
com.github.paolorotolo.appintro.AppIntroBaseFragment.onCreateView(Unknown Source)
android.support.v4.app.Fragment.performCreateView(Unknown Source)
android.support.v4.app.FragmentManagerImpl.moveToState(Unknown Source)
android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(Unknown Source)
android.support.v4.app.FragmentManagerImpl.moveToState(Unknown Source)
android.support.v4.app.BackStackRecord.executeOps(Unknown Source)
android.support.v4.app.FragmentManagerImpl.executeOps(Unknown Source)
android.support.v4.app.FragmentManagerImpl.executeOpsTogether(Unknown Source)
android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(Unknown Source)
android.support.v4.app.FragmentManagerImpl.execSingleAction(Unknown Source)
android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(Unknown Source)
android.support.v4.app.FragmentPagerAdapter.finishUpdate(Unknown Source)
android.support.v4.view.ViewPager.populate(Unknown Source)
android.support.v4.view.ViewPager.setCurrentItemInternal(Unknown Source)
android.support.v4.view.ViewPager.setCurrentItemInternal(Unknown Source)
android.support.v4.view.ViewPager.setCurrentItem(Unknown Source)
com.github.paolorotolo.appintro.AppIntroViewPager.setCurrentItem(Unknown Source)
com.github.paolorotolo.appintro.AppIntroBase$NextButtonOnClickListener.onClick(Unknown Source)
android.view.View.performClick(View.java:5637)
android.view.View$PerformClick.run(View.java:22429)
android.os.Handler.handleCallback(Handler.java:751)
android.os.Handler.dispatchMessage(Handler.java:95)
android.os.Looper.loop(Looper.java:154)
android.app.ActivityThread.main(ActivityThread.java:6169)
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778)
```

Two of the three logs I got came from a Nexus 6P, the third one is not known yet.
I should probably extend the Crash Log Sending Activity to get more data, so that I get the used device every time and not only if the users remember to add it.

**Comment by jcgruenhage** [comment]
There also has been a report of another phone, so this does not seem to be limited to the Nexus 6P.
I'm going to take a closer look on this, after fixing #7 and possibly do some automatic crash reporting.

**Comment by jcgruenhage** [comment]
I expect a quick'n'dirty hack for this for v0.9.5 (probably not before april), a simple try catch wrapper around the intro, that replaces the intro with a series of alert dialogs. @mgillespie what do you think about that?

**Comment by ghost** [comment]
Wondering if it's possible to simulate a low memory situation in the emulator, and/or perhaps try and reduce the size of the drawable resources?  (webp?)  I think I have seen this in some code I did once, where it's simply an error loading the drawable, perhaps a try-catch and just not show the drawable?

**Comment by jcgruenhage** [comment]
Just hiding the drawable would be even better, but that would rather need a fix in [apl-devs/AppInto](https://github.com/apl-devs/AppIntro) instead. 

I don't think simulating a low memory situation in an emulator would fix this, since the 6P has a lot of memory, and the people reporting this told me that they have even tried clearing the memory before that.

**Comment by ghost** [comment]
Related? https://github.com/apl-devs/AppIntro/issues/402

**Comment by jcgruenhage** [comment]
Probably. We should probably just scale down/compress all the images. Do you know anything about vector graphics? Replacing all those with SVGs would be the ideal solution, if file size really is the problem here.

**Comment by ghost** [comment]
Had a go at this.   https://github.com/jcgruenhage/PlayMusicExporter/pull/13

**Comment by jcgruenhage** [comment]
Well, we would need to make a new relese and send it out to the users, to check whether it has helped with anything but app size. Would you mind taking a look at #7? That is afaik the only thing that needs to be fixed now before v0.9.5, and I have a task for my programming class to finish that is due tomorrow, and and a (hard) exam on Wednesday..

**Comment by ghost** [comment]
I will try and take a look at the crash handier at the weekend. :-)

**Comment by jcgruenhage** [comment]
Okay, thank you :+1: 

It would be just adding a few lines to the automatic output to help determine which bugs may be device/rom specific.

**Comment by jcgruenhage** [comment]
I guess this issue is fixed now, if it still happens for people, I will reopen this.

**Comment by wrkindrk** [comment]
the issue still there! sent you an email for the log, which looks the same as the first post.

**Comment by jcgruenhage** [comment]
@mohaabed In v0.9.4, yes, but (unless you compiled the app yourself after the PR above was merged) hopefully not in v0.9.5, which I will release in a few minutes ^^

**Comment by jcgruenhage** [comment]
@mohaabed I have lost access to my key store because of a failed hard drive, which is one of the reasons, that I decided to move to F-Droid. In a few days, you will be able to download the app from F-Droid.
